# San Francisco Salaries

In [24]:
import pandas as pd

### Erstelle aus salaries.csv einen DataFrame

In [25]:
sal = pd.read_csv("./data/salaries.csv", na_values=['Not Provided'])

C:\Users\User\AppData\Local\Temp\ipykernel_28180\722447554.py:1: DtypeWarning: Columns (12) have mixed types. Specify dtype option on import or set low_memory=False.
  sal = pd.read_csv("./data/salaries.csv", na_values=['Not Provided'])


** Empfohlen: `head()` **

In [26]:
sal.head()

Id       EmployeeName                                        JobTitle  \
0   1     NATHANIEL FORD  GENERAL MANAGER-METROPOLITAN TRANSIT AUTHORITY   
1   2       GARY JIMENEZ                 CAPTAIN III (POLICE DEPARTMENT)   
2   3     ALBERT PARDINI                 CAPTAIN III (POLICE DEPARTMENT)   
3   4  CHRISTOPHER CHONG            WIRE ROPE CABLE MAINTENANCE MECHANIC   
4   5    PATRICK GARDNER    DEPUTY CHIEF OF DEPARTMENT,(FIRE DEPARTMENT)   

     BasePay  OvertimePay   OtherPay  Benefits   TotalPay  TotalPayBenefits  \
0  167411.18         0.00  400184.25       NaN  567595.43         567595.43   
1  155966.02    245131.88  137811.38       NaN  538909.28         538909.28   
2  212739.13    106088.18   16452.60       NaN  335279.91         335279.91   
3   77916.00     56120.71  198306.90       NaN  332343.61         332343.61   
4  134401.60      9737.00  182234.59       NaN  326373.19         326373.19   

   Year  Notes         Agency Status  
0  2011    NaN  San Francisco    NaN  
1  2011    NaN  San Francisco    NaN  
2  2011    NaN  San Francisco    NaN  
3  2011    NaN  San Francisco    NaN  
4  2011    NaN  San Francisco    NaN

### Wie viele Observations gibt es?

In [27]:
len(sal)

148654

### Was ist das arithmetische Mittel von `BasePay`?
Lösung: `66325.44884050643`

In [28]:
sal.BasePay.mean()

66325.44884050643

### Was ist der höchste Geldbetrag, der als `OvertimePay` vorhanden ist?
Lösung: `245131.88`

In [29]:
sal.OvertimePay.max()

245131.88

### Welchen Job hat JOSEPH DRISCOLL? (**Nicht** Joseph Driscoll).
Lösung:
```
CAPTAIN, FIRE SUPPRESSION
```

In [30]:
sal[sal.EmployeeName=="JOSEPH DRISCOLL"]['JobTitle']

24    CAPTAIN, FIRE SUPPRESSION
Name: JobTitle, dtype: object

### Was ist das Gesamtgehalt von JOSEPH DRISCOLL?
Lösung:
```
270324.91
```

In [31]:
sal[sal.EmployeeName=="JOSEPH DRISCOLL"]['TotalPay']

24    270324.91
Name: TotalPay, dtype: float64

### Wie lautet der Name der bestverdienensten Person?
Lösung: `NATHANIEL FORD`

In [32]:
sal[sal.TotalPay==sal.TotalPay.max()]['EmployeeName']

0    NATHANIEL FORD
Name: EmployeeName, dtype: object

### Wie lautet der Name der am schlechtest bezahlten Person?
Lösung: `Joe Lopez`

In [33]:
sal[sal.TotalPay==sal.TotalPay.min()]['EmployeeName']


148653    Joe Lopez
Name: EmployeeName, dtype: object

### Ermittle für die Jahre 2012-2014 den arithmetischen Mittelwert des `BasePay`s aller Personen.
Lösung:
```
Year
2012    65436.406857
2013    69630.030216
2014    66564.421924
```


In [63]:
sal[sal.Year.between(2012,2014)].groupby(by="Year").BasePay.mean()

Year
2012    65436.406857
2013    69630.030216
2014    66564.421924
Name: BasePay, dtype: float64

### Wie viele unterschiedliche Jobs gibt es?
Lösung: `2159`

In [50]:
sal.JobTitle.nunique()

2159

### Welche sind die Top 5 häufigsten Jobs und wie viele gibt es davon jeweils?
Lösung:
```
Transit Operator                7036
Special Nurse                   4389
Registered Nurse                3736
Public Svc Aide-Public Works    2518
Police Officer 3                2421
Name: JobTitle, dtype: int64
```


In [64]:
sal.groupby(by="JobTitle").size().sort_values(ascending=False)[:5]

JobTitle
Transit Operator                7036
Special Nurse                   4389
Registered Nurse                3736
Public Svc Aide-Public Works    2518
Police Officer 3                2421
dtype: int64

### Wie viele Jobs gibt es, die 2013 nur eine Person ausübte?
Lösung: `202`

In [104]:
(sal[sal.Year==2013].groupby(by="JobTitle").size()==1).sum()

KeyError: False

### Wie viele leute haben `'chief'` (caseinsensitive) im Jobnamen?
Lösung: `627`

In [107]:
sal[len(sal.JobTitle)==15].size()

KeyError: False

In [35]:
### Erstelle ein Histogramm 